Documentation/ApiOverview/ContentElements/CustomBackendPreview.rst

```diff
@@ -1,139 +1,98 @@
+:php:`\TYPO3\CMS\Backend\Preview\PreviewRendererInterface` which contains
 the following API methods:
 
-.. code-block:: php
-
-    /**
-     * Dedicated method for rendering preview header HTML for
-     * the page module only. Receives the the GridColumnItem
-     * that contains the record for which a preview header
-     * should be rendered and returned.
-     *
-     * @param GridColumnItem $item
-     * @return string
-     */
-    public function renderPageModulePreviewHeader(GridColumnItem $item): string;
-
-    /**
-     * Dedicated method for rendering preview body HTML for
-     * the page module only. Receives the the GridColumnItem
-     * that contains the record for which a preview should be
-     * rendered and returned.
-     *
-     * @param GridColumnItem $item
-     * @return string
-     */
-    public function renderPageModulePreviewContent(GridColumnItem $item): string;
-
-    /**
-     * Render a footer for the record to display in page module below
-     * the body of the item's preview.
-     *
-     * @param GridColumnItem $item
-     * @return string
-     */
-    public function renderPageModulePreviewFooter(GridColumnItem $item): string;
-
-    /**
-     * Dedicated method for wrapping a preview header and body
-     * HTML. Receives $item, an instance of GridColumnItem holding
-     * among other things the record, which can be used to determine
-     * appropriate wrapping.
-     *
-     * @param string $previewHeader
-     * @param string $previewContent
-     * @param GridColumnItem $item
-     * @return string
-     */
-    public function wrapPageModulePreview(string $previewHeader, string $previewContent, GridColumnItem $item): string;
-
-.. note::
-
-   Further methods are expected to be added in the future to support generic preview rendering, e.g. usages outside :php:`PageLayoutView`.
-
-Implementing these methods allows you to control the exact composition of the preview.
-
-This means assuming your :php:`PreviewRenderer` returns :html:`<h4>Header</h4>` from the header render method and :html:`<p>Body</p>` from
-the preview content rendering method and your wrapping method does :php:`return '<div>' . $previewHeader . $previewContent . '</div>';` then the
-entire output becomes :html:`<div><h4>Header</h4><p>Body</p></div>` when combined.
-
-Should you wish to reuse parts of the default preview rendering and only change, for example, the method that renders
-the preview body content, you can subclass :php:`\TYPO3\CMS\Backend\Preview\StandardContentPreviewRenderer` in your
-own :php:`PreviewRenderer` class - and selectively override the methods from the API displayed above.
-
+..  include:: /CodeSnippets/Manual/Backend/PreviewRendererInterface.rst.txt
 
-Configuring the implementation
-------------------------------
-
-Individual preview renderers can be defined by using one of the following two approaches:
+..  note::
+    Further methods are expected to be added in the future to support generic
```

Comment:
I would rather not talk about "the future". This is not very specific and has often not come true in the past

Reply:
Actually, I want to remove the note - if the interface is extended there will be another changelog and we can adjust this chapter accordingly. In v12 the PageContentPreviewRenderingEvent is introduced which seems to do something similar. Want to play a little bit with the event in the next time to see the differences.